bin/ch-run.c

```diff
@@ -6,70 +6,26 @@
 #define _GNU_SOURCE
 #include <argp.h>
 #include <assert.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <libgen.h>
-#include <sched.h>
-#include <semaphore.h>
 #include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/mman.h>
-#include <sys/mount.h>
-#include <sys/prctl.h>
-#include <sys/stat.h>
-#include <sys/syscall.h>
-#include <time.h>
 #include <unistd.h>
 
 #include "charliecloud.h"
 
-/** Constants and macros **/
-
-/* Host filesystems to bind. */
-#define USER_BINDS_MAX 10
-const char * DEFAULT_BINDS[] = { "/dev",
-                                 "/etc/passwd",
-                                 "/etc/group",
-                                 "/etc/hosts",
-                                 "/etc/resolv.conf",
-                                 "/proc",
-                                 "/sys",
-                                 NULL };
-
-/* Number of supplemental GIDs we can deal with. */
-#define SUPP_GIDS_MAX 128
 
-/* Maximum length of paths we're willing to deal with. (Note that
-   system-defined PATH_MAX isn't reliable.) */
-#define PATH_CHARS 4096
+/** Constants and macros **/
 
-/* Log the current UIDs. */
-#define LOG_IDS log_ids(__func__, __LINE__)
+/* Maximum number of user binds. */
 
-/* Environment variables used for --join parameters. */
+/* Environment variables used by --join parameters. */
 char * JOIN_CT_ENV[] =  { "OMPI_COMM_WORLD_LOCAL_SIZE",
                           "SLURM_STEP_TASKS_PER_NODE",
                           "SLURM_CPUS_ON_NODE",
                           NULL };
 char * JOIN_TAG_ENV[] = { "SLURM_STEP_ID",
                           NULL };
 
```

Comment:
I feel like these join environment variables should be moved to charliecloud.c. Keeping them bundled with the join macro definitions is much more intuitive and easier to read.

Reply:
They're in `ch-run.c` because I saw them as part of the user interface rather than containerization itself. Convincing?

Reply:
Future users may want a different set of variables, if any. E.g., if we put something into Slurm we won't want to be looking at OpenMPI environment variables in that context.